snowplow_tracker/emitters.py

```diff
@@ -43,13 +39,11 @@
 
 DEFAULT_MAX_LENGTH = 10
 PAYLOAD_DATA_SCHEMA = "iglu:com.snowplowanalytics.snowplow/payload_data/jsonschema/1-0-4"
+PROTOCOLS = {"http", "https"}
+METHODS = {"get", "post"}
 
-# contracts
-new_contract("protocol", lambda x: x == "http" or x == "https")
-
-new_contract("method", lambda x: x == "get" or x == "post")
-
-new_contract("function", lambda x: hasattr(x, "__call__"))
+SuccessCallback = Callable[[PayloadDictList], None]
+FailureCallback = Callable[[int, PayloadDictList], None]
```

Comment:
nit - would it make sense for these types to be defined in `snowplow_tracker.typing` as well?

Reply:
Right, moved all of these types to typing.py.